lib/Doctrine/Deprecations/Deprecation.php

```diff
@@ -72,15 +70,49 @@ class Deprecation
      */
     public static function trigger(string $package, string $link, string $message, ...$args): void
     {
-        // Do not trigger this deprecation if it is temporarily ignored,
-        // because it is expected to be called for 1 or more times.
-        if (array_key_exists($link, self::$temporarilyIgnoredLinks)) {
-            self::$temporarilyIgnoredLinks[$link]--;
+        if (array_key_exists($link, self::$ignoredLinks)) {
+            self::$ignoredLinks[$link]++;
+        } else {
+            self::$ignoredLinks[$link] = 1;
+        }
+
+        if (self::$deduplication === true && self::$ignoredLinks[$link] > 1) {
+            return;
+        }
+
+        // do not move this condition to the top, because we still want to
+        // count occcurences of deprecations even when we are not logging them.
+        if (self::$type === self::TYPE_NONE) {
+            return;
+        }
+
+        if (isset(self::$ignoredPackages[$package])) {
+            return;
+        }
+
+        $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2);
+
+        $message = sprintf($message, ...$args);
+
+        self::delegateTriggerToBackend($message, $backtrace, $link, $package);
+    }
+
+    /**
+     * @param mixed $args
+     */
+    public static function triggerIfCalledFromOutside(string $package, string $link, string $message, ...$args): void
+    {
+        $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2);
 
-            if (self::$temporarilyIgnoredLinks[$link] <= 0) {
-                unset(self::$temporarilyIgnoredLinks[$link]);
-            }
+        // "outside" means we assume that $package is currently installed as a
+        // dependency and the caller is not a file in that package.
+        // When $package is installed as a root package, then this deprecation
+        // is always ignored
+        if (strpos($backtrace[0]['file'], 'vendor/' . $package . '/') === false) {
```

Comment:
Apparently `vendor` can be configured: https://getcomposer.org/doc/06-config.md#vendor-dir

Here is a (hairy) way to detect it: https://github.com/symfony/phpunit-bridge/blob/f2985c4d17f5d2584c5bd66e13b2fd0e3a67416f/DeprecationErrorHandler/Deprecation.php#L312

But maybe that's overkill…

Reply:
I know its not perfect, but its a cheap way to do the check vs the one in phpunit bridge that does not have to care for performance that much.